modules/gdscript/gdscript_parser.cpp

```diff
@@ -3293,31 +3293,116 @@ GDScriptParser::TypeNode *GDScriptParser::parse_type(bool p_allow_void) {
 }
 
 #ifdef TOOLS_ENABLED
-static bool _in_codeblock(String p_line, bool p_already_in, int *r_block_begins = nullptr) {
-	int start_block = p_line.rfind("[codeblock]");
-	int end_block = p_line.rfind("[/codeblock]");
-
-	if (start_block != -1 && r_block_begins) {
-		*r_block_begins = start_block;
+enum DocLineState {
+	DOC_LINE_NORMAL,
+	DOC_LINE_IN_CODE,
+	DOC_LINE_IN_CODEBLOCK,
+};
+
+static String _process_doc_line(const String &p_line, const String &p_text, const String &p_space_prefix, DocLineState *r_state) {
+	ERR_FAIL_COND_V(r_state == nullptr, String());
```

Comment:
Could you make `r_state` a reference if it's never meant to be `null`?

Reply:
Done.